plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java

```diff
@@ -72,7 +71,7 @@ public static PreallocationType getPreallocationType(final Storage.ProvisioningT
             case FAT:
                 return PreallocationType.Full;
             default:
-                throw new NotImplementedException();
+                throw new NotImplementedProvisioningTypeException(String.format("type {} not defined as member-value of PreallocationType", provisioningType));
```

Comment:
Is this a new type of `String.format` use? I am missing a `%s` there.

Reply:
hm, i might be mixing python and java, will check.